Client Database System/Client Database System/Product and Services.cpp

```diff
@@ -8,13 +8,74 @@
 
 #include "Product and Services.hpp"
 #include <iostream>
+#include <fstream>
+#include <iomanip>
+#include <string>
 #include <vector>
 
 using namespace std;
 
-void AddPrice(vector<int>& priceVector){
+Products_and_Services::Products_and_Services(){
+    product = "No product";
+    serviceDescription = "No service description";
+}
+
+void Product_and_Services::AddPrice(vector<int>& priceVector){
     int userPrice;
     
     cin >> userPrice;
     priceVector.push_back(userPrice);
 }
+
+void Product_and_Services::ReadFromFile(vector<string>& productVector, vector<string>& descriptionVector){
```

Comment:
Yeah, these functions are quite confusing because we're dealing with external files.

So this requires the fstream library which has been included already. First what we do is create a reference to the input-stream by writing `ifstream inputStreamVariable;` This tells the program that inputStreamVariable is reading information from the file. But we have to tell the program where to get the information from.

To do this we write `inputStreamVariable.open("TextFile.txt");` and I already included a .txt file which is named "SaleRepresentativesFile.txt" so we would use that instead of "TextFile.txt". The open() function opens the file so that the program can read its contents.

Now we can read the inputs from the file, but we don't know how much information is in the file, so we use a while loop. And the argument is `(!inputStreamVariable.eof())` and `eof()` checks to see when it reaches the end of the file. Then, in the loop we get the input from the file and fill it into the vector.

Reply:
And I forgot that we need to close the file before we finish. So after the loop we would write `inputStreamVariable.close()` to close the file from the program.